src/NLog.Extensions.Logging/Extensions/ConfigureExtensions.cs

```diff
@@ -83,5 +111,48 @@ public static LoggingConfiguration ConfigureNLog(this ILoggerFactory loggerFacto
             LogManager.Configuration = config;
             return config;
         }
+
+        /// <summary>
+        /// Factory method for <see cref="NLogLoggerProvider"/>
+        /// </summary>
+        /// <param name="nlogProvider"></param>
+        /// <param name="configurationSection">Microsoft Extension Configuration</param>
+        /// <returns></returns>
+        public static NLogLoggerProvider ConfigureNLogProvider(this NLogLoggerProvider nlogProvider, IConfigurationSection configurationSection)
+        {
+            if (configurationSection == null)
+                return nlogProvider;
+
+            var configProps = nlogProvider.Options.GetType().GetProperties(BindingFlags.Instance | BindingFlags.Public).Where(p => p.SetMethod?.IsPublic == true).ToDictionary(p => p.Name, StringComparer.OrdinalIgnoreCase);
+            foreach (var configValue in configurationSection.GetChildren())
+            {
+                if (configProps.TryGetValue(configValue.Key, out var propertyInfo))
+                {
+                    try
+                    {
+                        var result = Convert.ChangeType(configValue.Value, propertyInfo.PropertyType);
+                        propertyInfo.SetMethod.Invoke(nlogProvider.Options, new[] { result });
+                    }
+                    catch (Exception ex)
+                    {
+                        InternalLogger.Warn(ex, "NLogProviderOptions: Property {0} could not be assigned value: {1}", configValue.Key, configValue.Value);
+                    }
+                }
+            }
+
+            return nlogProvider;
+        }
+
+        private static NLogLoggerProvider CreateNLogProvider(IConfiguration configuration)
+        {
+            var provider = new NLogLoggerProvider(new NLogProviderOptions());
+            if (configuration != null)
+            {
+                // TODO ConfigSettingLayoutRenderer.DefaultConfiguration = configuration;
```

Comment:
is this TODO for this PR, or do we need a Github issue?

Reply:
Next PR. You just requested me to split this into more PRs